ts/Series/HollowCandlestick/HollowCandlestickSeries.ts

```diff
@@ -0,0 +1,344 @@
+         * @product highstock
+         */
+        upLineColor: palette.positiveColor
+
+    } as HollowCandlestickSeriesOptions);
+
+    /* *
+     *
+     * Properties
+     *
+     * */
+    public data: Array<HollowCandlestickPoint> = void 0 as any;
+
+    public options: HollowCandlestickSeriesOptions = void 0 as any;
+
+    public points: Array<HollowCandlestickPoint> = void 0 as any;
+
+    /* *
+     *
+     * Functions
+     *
+     * */
+
+    /**
+     * Add fill attribute for each points based on
+     * the previously calculated value.
+     *
+     * @function Highcharts.seriesTypes.hollowcandlestick#pointAttribs
+     *
+     * @param {HollowCandlestickPoint} point
+     *        Point to which we are adding attributes.
+     * @param {StatesOptionsKey} state
+     *        Current point state.
+     *
+     * @return {SVGAttributes}
+     *
+     */
+    public pointAttribs(
+        point: HollowCandlestickPoint,
+        state?: StatesOptionsKey
+    ): SVGAttributes {
+        let attribs = super.pointAttribs.call(this, point, state);
+
+        attribs.fill = point.candleFill || attribs.fill;
+        attribs.stroke = point.color || attribs.stroke;
+        return attribs;
+    }
+
+    /**
+     * Translate from value to pixel as a base method and loop through points
+     * in order to calculate the fill.
+     *
+     * @function Highcharts.seriesTypes.hollowcandlestick#translate
+     *
+     * @return {void}
+     *
+     */
+    public translate(): void {
+        // Run the base method.
+        super.translate.apply(this);
+
+        const series = this,
+            points = series.points;
+
+        if (points && points.length) {
+            // The first point always without fill,
+            // because of the lack of the previous point.
+            points[0].candleFill = 'transparent';
+
+            for (let i = 1; i < points.length; i++) {
+                const point = points[i],
+                    previousPoint = points[i - 1];
+
+                point.candleFill = point.getPointFill(previousPoint);
+                point.color = point.getLineColor(previousPoint);
+            }
+        }
+    }
+
+    /**
+     * In order to distinguish two types of bearish points,
+     * add the class to a point if needed.
+     *
+     * @function Highcharts.seriesTypes.hollowcandlestick#drawPoints
+     *
+     * @return {void}
+     *
+     */
+    public drawPoints(): void {
+        // Run the base method.
+        super.drawPoints.apply(this);
+
+        const series = this,
+            points = series.points;
+
+        if (points && points.length) {
+            for (let i = 1; i < points.length; i++) {
+                const point = points[i],
+                    previousPoint = points[i - 1];
+                if (point.graphic && point.close > previousPoint.close && point.candleFill !== 'transparent') {
+                    point.graphic.addClass(point.getClassName() + '-bearish-up', true);
```

Comment:
With overwriting `translate()` and `drawPoints()` we need to loop over all points 3x more than the basic version. Would it make sense to calculate the type (bearish/bullish etc) of the candlestick just once, and let rendering to the rest? For example instead of overwriting `drawPoints()` you could overwrite `point.getClassName()` and add the class there.